### DISCUSSION TITLE
Packing with unlimited "bins"

### DISCUSSION THREAD

**Discussion by luxaritas** [TimefoldAI/timefold-solver#1266] [General]
I currently have a problem where I'm trying to optimize packing tasks into batch jobs, and each job has some limits on the total resources the tasks can use. I want to minimize the number of jobs (bins), but allow it to have a potentially unlimited upper-bound. The natural way to model this is for each job to be a planning fact and each task to be a planning entity, with the job to be selected by the solver (my original intuition was to do it the other way around, with the list of tasks being a list variable, but from my reading it sounds like that is not the right way to go because that encodes order?). However, it is my understanding that in this case, all the facts/jobs would need to be instantiated ahead of time, rather than being dynamically instantiated. I could of course instantiate an arbitrarily large number of jobs to schedule into, but that seems a bit silly.

Any recommendations? Thanks!

**Comment by triceo** [comment]
Hello @luxaritas. 

> I could of course instantiate an arbitrarily large number of jobs to schedule into, but that seems a bit silly.

I agree it is "a bit silly", but it's probably the best thing to do. We do not allow new entities to be added during solving, and therefore doing this dynamically is not possible.

**Reply by luxaritas** [reply]
Alright, thank you!
